File: behaviorial/command/python/invoker.py
class Button:
    def __init__(self, label):
        self.label = label
        self.command = None

    def set_command(self, command):
        self.command = command

    def press(self):
        if self.command:
            print(f'Button {self.label} pressed')
            self.command.execute()

    def undo(self):
        if self.command:
            print(f'Button {self.label} undo')
            self.command.undo()


class RemoteControl:
    def __init__(self):
        self.buttons = {}

    def add_button(self, label, button):
        self.buttons[label] = button

    def press_button(self, label):
        button = self.buttons.get(label)
        if button:
            button.press()
        else:
            print(f'Button {label} not found')

    def undo_button(self, label):
        button = self.buttons.get(label)
        if button:
            button.undo()
        else:
            print(f'Button {label} not found')

File: behaviorial/command/typescript/invoker.ts
import { Command } from './interface';

export class Button {
	private command: Command | null = null;
	private label: string;

	constructor(label: string) {
		this.label = label;
	}

	setCommand(command: Command): void {
		this.command = command;
	}

	press(): void {
		if (this.command) {
			console.log(`Button ${this.label} pressed`);
			this.command.execute();
		}
	}

	undo(): void {
		if (this.command) {
			console.log(`Button ${this.label} undo`);
			this.command.undo();
		}
	}
}

export class RemoteControl {
	private buttons: Map<string, Button> = new Map();

	addButton(label: string, button: Button): void {
		this.buttons.set(label, button);
	}

	pressButton(label: string): void {
		const button = this.buttons.get(label);
		if (button) {
			button.press();
		} else {
			console.log(`Button ${label} not found`);
		}
	}

	undoButton(label: string): void {
		const button = this.buttons.get(label);
		if (button) {
			button.undo();
		} else {
			console.log(`Button ${label} not found`);
		}
	}
}
